src/image.rs

```diff
@@ -591,23 +619,132 @@ impl BuildOptions {
 #[derive(Default)]
 pub struct BuildOptionsBuilder {
     path: String,
-    params: HashMap<&'static str, String>,
+
+    build_params: BuildParams,
+
     skip_gzip: bool,
 }
 
 impl BuildOptionsBuilder {
     /// path is expected to be a file path to a directory containing a Dockerfile
     /// describing how to build a Docker image
     pub(crate) fn new<S>(path: S) -> Self
-    where
-        S: Into<String>,
+        where
+            S: Into<String>,
     {
         BuildOptionsBuilder {
             path: path.into(),
             ..Default::default()
         }
     }
 
+    pub fn set_skip_gzip(
+        &mut self,
+        skip_gzip: bool,
+    ) -> &mut Self {
+        self.skip_gzip = skip_gzip;
+        self
+    }
+
+    pub fn dockerfile<P>(
```

Comment:
The interface for settings a single API param is copied from `BuildParams` to not break the original `BuildOptionsBuilder` type. The need for more simple `BuildParams` type arises from the fact that new `build_from_raw_parts` function doesn't require `path` or `skip_gzip` arguments. I also believe that this part of crate should be redesigned, as it contains duplicate data and a code-copy in some places, but this is out of scope of this story.